Book-A-Desk.Domain/CommandHandler.Reservations.fs

```diff
@@ -5,12 +5,38 @@ open Book_A_Desk.Domain
 open Book_A_Desk.Domain.Events
 open Book_A_Desk.Domain.Reservation
 open Book_A_Desk.Domain.Reservation.Domain
+open Book_A_Desk.Domain.Reservation.Events
 open Book_A_Desk.Domain.Reservation.Commands
+open Book_A_Desk.Domain.Reservation.Commands.Instructions.Reservation
 
 type ReservationsCommandHandler =
     {
         Handle: ReservationCommand -> Result<unit,string>
     }
+    
+module ReservationsCommandHandlerUsingInstructions =
```

Comment:
This is the interesting part. To understand what this does, think that everytime you see a "!", we send a command to an interpreter with our defined input and response. So:

`let! offices = getOffices`

Sends the GetOffices command to the interpreter, who returns the list of offices, however that happens, whether it happens in an async manner, it doesn't matter to the domain.

Note: this code does the same thing as ReservationsCommandHandler.handle, I just didn't bother using the command executor and the dependencies to show a simple example.